src/servant/tt_002/3678.c

```diff
@@ -4,12 +4,26 @@
 
 extern s32 s_ServantId;
 extern u16 D_us_801722E8[];
+// During cleanup, rename this.  May not actually be this familiar, unknown
+// where it's set
 extern Entity thisFamiliar;
 extern s32 s_zPriority;
 extern FamiliarStats s_FaerieStats;
 extern s32 D_us_8017931C;
 extern s32 D_us_80179320;
 
+// this may actually be a multi dimensional array instead of a struct
+typedef struct {
+    s16 unk0;
+    s16 unk2;
+} UnkFaerieStruct;
+
+extern UnkFaerieStruct D_us_80172368[];
```

Comment:
Yeah, you are probably right.  I didn't notice that and the function is definitely doing stuff with animation frame data, so that seems very reasonable.  I'll clean it up on my next function.